src/jdk.javadoc/share/classes/jdk/javadoc/internal/doclets/toolkit/util/Utils.java

```diff
@@ -1181,9 +1180,13 @@ public TypeElement getSuperClass(TypeElement te) {
         return asTypeElement(superclass);
     }
 
+    private boolean checkType(TypeElement te) {
+        return isInterface(te) || typeUtils.isSameType(te.asType(), getObjectType())
+                || isAnnotationType(te);
+    }
+
     public TypeElement getFirstVisibleSuperClassAsTypeElement(TypeElement te) {
```

Comment:
If you were to add a comment here, I would note that the first visible superclass explicitly does NOT include `java.lang.Object`.  There's an underlying policy in javadoc to ignore some types like that which are "too prolific". IIRC, for a while, this included CORBA's Object as well.